data/transactions/logic/eval_test.go

```diff
@@ -598,243 +606,52 @@ func TestDivZero(t *testing.T) {
-			ops, err := AssembleStringWithVersion(`byte 0x10101010`, v)
-			require.NoError(t, err)
-			sb := strings.Builder{}
-			err = Check(ops.Program, defaultEvalParams(&sb, nil))
-			require.NoError(t, err)
-			sb = strings.Builder{}
-			pass, err := Eval(ops.Program, defaultEvalParams(&sb, nil))
-			if pass {
-				t.Log(hex.EncodeToString(ops.Program))
-				t.Log(sb.String())
-			}
-			require.Error(t, err)
-			require.False(t, pass)
-			isNotPanic(t, err)
-		})
-	}
+	testPanics(t, "byte 0x10101010", 1)
 }
 
 func TestStackEmpty(t *testing.T) {
 	t.Parallel()
-	for v := uint64(1); v <= AssemblerMaxVersion; v++ {
-		t.Run(fmt.Sprintf("v=%d", v), func(t *testing.T) {
-			ops, err := AssembleStringWithVersion(`int 1
-int 1
-pop
-pop`, v)
-			require.NoError(t, err)
-			err = Check(ops.Program, defaultEvalParams(nil, nil))
-			require.NoError(t, err)
-			sb := strings.Builder{}
-			pass, err := Eval(ops.Program, defaultEvalParams(&sb, nil))
-			if pass {
-				t.Log(hex.EncodeToString(ops.Program))
-				t.Log(sb.String())
-			}
-			require.Error(t, err)
-			require.False(t, pass)
-			isNotPanic(t, err)
-		})
-	}
+	testPanics(t, "int 1; int 1; pop; pop", 1)
 }
 
 func TestArgTooFar(t *testing.T) {
 	t.Parallel()
-	for v := uint64(1); v <= AssemblerMaxVersion; v++ {
-		t.Run(fmt.Sprintf("v=%d", v), func(t *testing.T) {
-			ops, err := AssembleStringWithVersion(`arg_1
-btoi`, v)
-			require.NoError(t, err)
-			err = Check(ops.Program, defaultEvalParams(nil, nil))
-			require.NoError(t, err) // TODO: Check should know the type stack was wrong
-			sb := strings.Builder{}
-			var txn transactions.SignedTxn
-			txn.Lsig.Logic = ops.Program
-			txn.Lsig.Args = nil
-			pass, err := Eval(ops.Program, defaultEvalParams(&sb, &txn))
-			if pass {
-				t.Log(hex.EncodeToString(ops.Program))
-				t.Log(sb.String())
-			}
-			require.Error(t, err)
-			require.False(t, pass)
-			isNotPanic(t, err)
-		})
-	}
+	testPanics(t, "arg_1; btoi", 1)
 }
 
 func TestIntcTooFar(t *testing.T) {
 	t.Parallel()
-	for v := uint64(1); v <= AssemblerMaxVersion; v++ {
-		t.Run(fmt.Sprintf("v=%d", v), func(t *testing.T) {
-			ops, err := AssembleStringWithVersion(`intc_1`, v)
-			require.NoError(t, err)
-			err = Check(ops.Program, defaultEvalParams(nil, nil))
-			require.NoError(t, err) // TODO: Check should know the type stack was wrong
-			sb := strings.Builder{}
-			var txn transactions.SignedTxn
-			txn.Lsig.Logic = ops.Program
-			txn.Lsig.Args = nil
-			pass, err := Eval(ops.Program, defaultEvalParams(&sb, &txn))
-			if pass {
-				t.Log(hex.EncodeToString(ops.Program))
-				t.Log(sb.String())
-			}
-			require.Error(t, err)
-			require.False(t, pass)
-			isNotPanic(t, err)
-		})
-	}
+	testPanics(t, "intc_1", 1)
 }
 
 func TestBytecTooFar(t *testing.T) {
 	t.Parallel()
-	for v := uint64(1); v <= AssemblerMaxVersion; v++ {
-		t.Run(fmt.Sprintf("v=%d", v), func(t *testing.T) {
-			ops, err := AssembleStringWithVersion(`bytec_1
-btoi`, v)
```

Comment:
is `btoi` not needed in the test?

Reply:
Good catch.  The code still panics without it, but it's a better test with it in.  We want the program to have be plausibly accepted, so we're really testing that reading the bytec_1 fails.  In fact, I think "int 1; return" is an even better way to finish, so that we have the best possible change of accepting if bytec_1 doesn't fail.